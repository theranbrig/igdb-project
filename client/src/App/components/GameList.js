import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const GameList = ({ gameList }) => (
  <>
    {gameList.map(item => (
      <div className="game-link" key={item.gameId}>
        <Link
          to={{
            pathname: `/game/${item.gameId}/${item.platform}`,
          }}
        >
          <p className="nes-text is-primary">{item.name}</p>
        </Link>
      </div>
    ))}
  </>
);

GameList.propTypes = {
  gameList: PropTypes.array.isRequired,
};

export default GameList;
