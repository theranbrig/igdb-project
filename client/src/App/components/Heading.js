import React from 'react';
import { HeadingStyles } from '../styles/MainStyles';

const Heading = () => (
	<HeadingStyles>
		<h1>
			<i className="nes-icon nes-mario is-small" /> NintenDB{' '}
			<i className="nes-icon nes-kirby is-small" />
		</h1>
	</HeadingStyles>
);

export default Heading;
