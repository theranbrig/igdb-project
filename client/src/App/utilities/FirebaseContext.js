import React, { useState } from 'react';
import firebase from 'firebase';
import PropTypes from 'prop-types';

export const FirebaseContext = React.createContext();

const config = {
  apiKey: process.env.REACT_APP_API_KEY,
  authDomain: process.env.REACT_APP_AUTH_DOMAIN,
  databaseURL: process.env.REACT_APP_DATABASE_URL,
  projectId: process.env.REACT_APP_PROJECT_ID,
  storageBucket: process.env.REACT_APP_STORAGE_BUCKET,
  messagingSenderId: process.env.REACT_APP_MESSAGING_SENDER_ID,
};

firebase.initializeApp(config);

const FirebaseProvider = props => {
  const [error, setError] = useState(null);
  const [authUser, setAuthUser] = useState(null);
  const [loading, setLoading] = useState(false);
  const [liked, setLiked] = useState(false);
  const [savedGames, setSavedGames] = useState([]);
  const [likedGame, setLikedGame] = useState(null);

  const db = firebase.firestore();

  const handleSignUp = async (email, password) => {
    setLoading(true);
    const user = await firebase
      .auth()
      .createUserWithEmailAndPassword(email, password)
      .then(setLoading(false))
      .catch(function(error) {
        setError(error.message);
        setLoading(false);
      });
    setAuthUser(user);
  };

  const handleSignIn = async (email, password) => {
    setLoading(true);
    const user = await firebase
      .auth()
      .signInWithEmailAndPassword(email, password)
      .then(setLoading(false))
      .catch(function(error) {
        setError(error.message);
        setLoading(false);
      });
    setAuthUser(user);
  };

  const handleLogout = async () => {
    setLoading(true);
    await firebase.auth().signOut();
    setAuthUser(null);
    setLoading(false);
  };

  firebase.auth().onAuthStateChanged(function(user) {
    if (user) {
      setAuthUser(user);
    } else {
      console.log('No User Found');
    }
  });

  const checkGameSave = async (userId, gameId) => {
    setLiked(false);
    await db
      .collection('votes')
      .where('userId', '==', userId)
      .where('gameId', '==', gameId)
      .get()
      .then(function(querySnapshot) {
        querySnapshot.forEach(function(doc) {
          if (doc.data()) {
            setLiked(true);
            setLikedGame(doc.data().likeId);
          }
        });
      })
      .catch(function(error) {
        console.log('Error getting documents: ', error);
      });
  };

  const handleGameSave = async (userId, gameId, name, platform) => {
    await db
      .collection('votes')
      .add({
        userId,
        gameId,
        name,
        platform,
      })
      .then(async function(docRef) {
        await checkGameSave(userId, docRef.id);
      })
      .catch(function(error) {
        console.error('Error adding document: ', error);
      });
  };

  const handleGameUnsave = async (userId, gameId) => {
    await db
      .collection('votes')
      .where('userId', '==', userId)
      .where('gameId', '==', gameId)
      .get()
      .then(function(querySnapshot) {
        querySnapshot.forEach(function(doc) {
          doc.ref.delete();
        });
      })
      .catch(function(error) {
        console.log('Error getting documents: ', error);
      });
  };

  const checkSavedUserGames = async userId => {
    const tempSavedGames = [];
    await db
      .collection('votes')
      .where('userId', '==', userId)
      .get()
      .then(querySnapshot => {
        querySnapshot.forEach(function(doc) {
          if (doc.data()) {
            tempSavedGames.push(doc.data());
          }
        });
      })
      .catch(function(error) {
        console.log('Error getting documents: ', error);
      });
    setSavedGames(tempSavedGames);
  };

  return (
    <FirebaseContext.Provider
      value={{
        error,
        authUser,
        handleSignUp,
        handleSignIn,
        handleLogout,
        loading,
        handleGameSave,
        checkGameSave,
        checkSavedUserGames,
        liked,
        savedGames,
        setLoading,
        likedGame,
        handleGameUnsave,
        setLiked,
      }}
    >
      {props.children}
    </FirebaseContext.Provider>
  );
};

FirebaseProvider.propTypes = {
  children: PropTypes.node.isRequired,
};

export default FirebaseProvider;
