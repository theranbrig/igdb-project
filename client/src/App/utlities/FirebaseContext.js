import React, { useState } from 'react';
import firebase from 'firebase';

export const FirebaseContext = React.createContext();

const config = {
	apiKey: process.env.REACT_APP_API_KEY,
	authDomain: process.env.REACT_APP_AUTH_DOMAIN,
	databaseURL: process.env.REACT_APP_DATABASE_URL,
	projectId: process.env.REACT_APP_PROJECT_ID,
	storageBucket: process.env.REACT_APP_STORAGE_BUCKET,
	messagingSenderId: process.env.REACT_APP_MESSAGING_SENDER_ID
};

firebase.initializeApp(config);

const FirebaseProvider = props => {
	const [error, setError] = useState(null);
	// Auth User = authUser.uid /
	const [authUser, setAuthUser] = useState(null);
	const [loading, setLoading] = useState(false);
	const [liked, setLiked] = useState(false);
	const [savedGames, setSavedGames] = useState([]);

	const db = firebase.firestore();

	const handleSignUp = async (email, password) => {
		setLoading(true);
		const user = await firebase
			.auth()
			.createUserWithEmailAndPassword(email, password)
			.then(() => db.collection('users').add({ email }))
			.then(docRef => {
				db.collection('users')
					.doc(docRef.id)
					.update({ userId: docRef.id });
				setLoading(false);
			})
			.catch(function(error) {
				setError(error.message);
				setLoading(false);
			});
		setAuthUser(user);
	};

	const handleSignIn = async (email, password) => {
		setLoading(true);
		const user = await firebase
			.auth()
			.signInWithEmailAndPassword(email, password)
			.then(setLoading(false))
			.catch(function(error) {
				setError(error.message);
				setLoading(false);
			});
		setAuthUser(user);
	};

	const handleLogout = async () => {
		setLoading(true);
		await firebase.auth().signOut();
		setAuthUser(null);
		setLoading(false);
	};

	firebase.auth().onAuthStateChanged(function(user) {
		if (user) {
			setAuthUser(user);
		} else {
			console.log('No User Found');
		}
	});

	const handleGameSave = async (userId, gameId, name, platform) => {
		const like = await db
			.collection('votes')
			.add({
				userId,
				gameId,
				name,
				platform
			})
			.then(function(docRef) {
				console.log('Document written with ID: ', docRef.id);
			})
			.catch(function(error) {
				console.error('Error adding document: ', error);
			});
	};

	const checkGameSave = async (userId, gameId) => {
		setLiked(false);
		const saveCheck = await db
			.collection('votes')
			.where('userId', '==', userId)
			.where('gameId', '==', gameId)
			.get()
			.then(function(querySnapshot) {
				querySnapshot.forEach(function(doc) {
					if (doc.data()) {
						setLiked(true);
					}
				});
			})
			.catch(function(error) {
				console.log('Error getting documents: ', error);
			});
	};

	const checkSavedUserGames = async userId => {
		const tempSavedGames = [];
		const response = await db
			.collection('votes')
			.where('userId', '==', userId)
			.get()
			.then(querySnapshot => {
				querySnapshot.forEach(function(doc) {
					if (doc.data()) {
						tempSavedGames.push(doc.data());
					}
				});
			})
			.catch(function(error) {
				console.log('Error getting documents: ', error);
			});
		setSavedGames(tempSavedGames);
	};

	return (
		<FirebaseContext.Provider
			value={{
				error,
				authUser,
				handleSignUp,
				handleSignIn,
				handleLogout,
				loading,
				handleGameSave,
				checkGameSave,
				checkSavedUserGames,
				liked,
				savedGames
			}}>
			{props.children}
		</FirebaseContext.Provider>
	);
};

export default FirebaseProvider;
