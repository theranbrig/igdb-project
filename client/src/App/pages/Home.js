import styled from 'styled-components';
import React from 'react';
import Form from '../components/Form';
import Heading from '../components/Heading';
import NavBar from '../components/NavBar';

const MainStyles = styled.div`
	.App {
		margin: 20px;
		min-height: 80vh;
		display: grid;
		grid-template-rows: 1fr;
		grid-template-columns: 1fr;
		min-height: 80vh;
		align-items: center;
		justify-items: center;
		margin: 10vh auto;
		width: 90%;
		.title {
			font-family: 'Press Start 2p';
		}
	}
	h1,
	i {
		padding: 15px 0 0;
		font-family: 'Press Start 2p';
		font-size: 5rem;
		text-align: center;
	}
	p {
		font-family: 'Press Start 2p';
	}
	.loading-screen {
		display: grid;
		grid-template-rows: 1fr;
		grid-template-columns: 1fr;
		min-height: 80vh;
		align-items: center;
		justify-items: center;
		margin: 10vh auto;
	}
	.icon-list {
		margin: 30px;
		i {
			margin: 20px !important;
		}import Heading from './Heading';
import NavBar from '../components/NavBar';

	}
`;

const Home = () => (
  <MainStyles>
    <NavBar />
    <div className="App nes-container is-dark is-rounded">
      <Heading />
      <Form />
      <section className="icon-list">
        <i className="nes-ash" />
        <i className="nes-pokeball" />
        <i className="nes-bulbasaur" />
        <i className="nes-charmander" />
        <i className="nes-squirtle" />
      </section>
    </div>
  </MainStyles>
);

export default Home;
