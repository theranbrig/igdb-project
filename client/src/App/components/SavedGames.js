import React, { useContext, useEffect } from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { FirebaseContext } from '../utilities/FirebaseContext';
import GameList from './GameList';
import Spinner from './Spinner';

const GameListStyles = styled.div`
  .games-list {
    display: grid;
    grid-template-columns: 1fr 1fr;
    grid-template-rows: 1fr 1fr;
    grid-gap: 20px;
    justify-items: center;
    font-family: 'Press Start 2p';
    width: 90%;
    margin-left: 5%;
    @media (max-width: 700px) {
      grid-template-columns: 1fr;
      grid-template-rows: 1fr;
    }
  }
  .game-list {
    text-align: center;
    width: 100%;
    a {
      text-decoration: none;
    }
  }
  .game-link {
    padding: 10px;
  }
  h1 {
    font-family: 'Press Start 2p';
    text-align: center;
    padding: 50px 0;
    @media (max-width: 700px) {
      font-size: 1.2rem;
    }
  }
`;

const SavedGames = ({ userId, user }) => {
  const { checkSavedUserGames, savedGames, setLoading, loading } = useContext(FirebaseContext);

  useEffect(() => {
    setLoading(true);
    checkSavedUserGames(userId);
    setLoading(false);
  }, [savedGames.length]);

  // Individual Game System Filters
  const nesGames = savedGames.filter(game => game.platform === 18);
  const snesGames = savedGames.filter(game => game.platform === 19);
  const sixtyFourGames = savedGames.filter(game => game.platform === 4);
  const gameCubeGames = savedGames.filter(game => game.platform === 21);

  return (
    <GameListStyles>
      {loading ? (
        <Spinner loading={loading} />
      ) : (
        <>
          <h1>{user}'s List of Games</h1>
          {savedGames.length ? (
            <div className="games-list">
              <div className="game-list nes-container is-rounded">
                <h2>NES Games</h2>
                <GameList gameList={nesGames} />
              </div>
              <div className="game-list nes-container is-rounded">
                <h2>SNES Games</h2>
                <GameList gameList={snesGames} />
              </div>
              <div className="game-list nes-container is-rounded">
                <h2>64 Games</h2>
                <GameList gameList={sixtyFourGames} />
              </div>
              <div className="game-list nes-container is-rounded">
                <h2>Game Cube Games</h2>
                <GameList gameList={gameCubeGames} />
              </div>
            </div>
          ) : (
            <h1>No Games Found. Start Adding Today.</h1>
          )}
        </>
      )}
    </GameListStyles>
  );
};

SavedGames.propTypes = {
  userId: PropTypes.string.isRequired,
  user: PropTypes.string.isRequired,
};

export default SavedGames;
