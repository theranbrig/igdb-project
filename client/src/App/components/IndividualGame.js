import React, { useState, useContext, useEffect } from 'react';
import { Link } from 'react-router-dom';
import ImagePlaceHolder from '../static/images/Image-Holder.jpg';
import FavoriteGameButton from './FavoriteGameButton';
import Carousel from './Carousel';
import { FirebaseContext } from '../utlities/FirebaseContext';

const IndividualGame = ({ data, platformId }) => {
	const [loved, setLoved] = useState(false);

	const toggleHeart = () => {
		setLoved(!loved);
	};

	const { authUser, checkGameSave } = useContext(FirebaseContext);

	useEffect(() => {
		checkGameSave(authUser.uid, data.id);
	}, []);

	return (
		<div className="main-game-content">
			<div className="nes-container with-title is-rounded is-centered">
				<div className="title-area">
					<h1>{data.name}</h1>
					<div className="loved-box">
						{authUser ? (
							<FavoriteGameButton
								toggleHeart={toggleHeart}
								loved={loved}
								gameId={data.id}
								name={data.name}
								platformId={platformId}
							/>
						) : (
							<Link to="/login">
								<a className="nes-text is-success">Sign In To Save</a>
							</Link>
						)}
						<p>IGDB Rating: {Math.floor(data.rating)}%</p>
					</div>
				</div>
				<div className="top-splash">
					<div className="nes-container is-dark is-rounded">
						{data.cover ? (
							<img
								src={`https://images.igdb.com/igdb/image/upload/t_cover_big/${
									data.cover.image_id
								}.jpg`}
								alt={data.cover.image_id}
							/>
						) : (
							<img src={ImagePlaceHolder} alt="No Cover Found" />
						)}
					</div>
					<div className="nes-container is-dark is-rounded with-title summary-box">
						<p className="title">Summary</p>
						{data.summary ? <p>{data.summary}</p> : <p>No Summary Found</p>}
					</div>
				</div>
				<div className="screenshots">
					{data.screenshots && <Carousel pictures={data.screenshots} />}
				</div>
			</div>
		</div>
	);
};

export default IndividualGame;
