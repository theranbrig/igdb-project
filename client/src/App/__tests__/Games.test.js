import React from 'react';
import TestRenderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
<<<<<<< HEAD
import shallow from 'enzyme';
import Heading from '../components/Heading';
import Home from '../pages/Home';
import { FirebaseContext } from '../utilities/FirebaseContext';
import GameList from '../components/GameList';
import SavedGames from '../components/SavedGames';
import IndividualGame from '../components/IndividualGame';

const authUser = { email: 'fake@fake.com' };
const nonAuthUser = null;
const platform = 18;
const data = {
  id: 79335,
  cover: { id: 55536, image_id: 'tfgefdirmoonz50hyjst' },
  genres: [{ id: 25, name: "Hack and slash/Beat 'em up" }],
  involved_companies: [{ id: 60798, company: [Object] }],
  name: 'Double Dragon III: The Sacred Stones',
  platforms: [{ id: 18, name: 'Nintendo Entertainment System (NES)' }, { id: 41, name: 'Wii U' }],
  rating: 81.32654202545959,
  release_dates: [{ id: 138927, y: 1990 }, { id: 138928, y: 1991 }, { id: 138929, y: 2015 }, { id: 138930, y: 2016 }],
  screenshots: [{ id: 157096, image_id: 'gl1e7emzpe5b7xvbfgp3' }],
  summary:
    'Billy, Jimmy, and the whole Double Dragon gang fight for the Sacred Stones to save in the world in yet another beat-em-up.',
};
=======
import { FirebaseContext } from '../utilities/FirebaseContext';
import GameList from '../components/GameList';
import SavedGames from '../components/SavedGames';

const authUser = { email: 'fake@fake.com' };
const nonAuthUser = null;
>>>>>>> d6254eeabc601f61666a349ccec4f094319235ca

const gameList = [{ gameId: 79335, name: 'Double Dragon III: The Sacred Stones', platform: 18 }];

it('renders the gameList item', () => {
<<<<<<< HEAD
  const element = TestRenderer.create(
    <Router>
      <FirebaseContext.Provider value={{ authUser: nonAuthUser }}>
        <GameList gameList={gameList} />
      </FirebaseContext.Provider>
    </Router>
  ).toJSON();
  expect(element).toMatchSnapshot();
});

it('renders the saved games component', () => {
  const element = TestRenderer.create(
    <Router>
      <FirebaseContext.Provider value={{ savedGames: gameList }}>
        <SavedGames userId="123456789" user="fake@fake.com" />
      </FirebaseContext.Provider>
    </Router>
  ).toJSON();
  expect(element).toMatchSnapshot();
});

it('renders the saved games component', () => {
  const element = shallow(<IndividualGame data={data} />).toJSON();
=======
	const element = TestRenderer.create(
		<Router>
			<FirebaseContext.Provider value={{ savedGames: gameList }}>
				<SavedGames userId="12345" gameId="123455" />
			</FirebaseContext.Provider>
		</Router>
	).toJSON();
	expect(element).toMatchSnapshot();
>>>>>>> d6254eeabc601f61666a349ccec4f094319235ca
});
