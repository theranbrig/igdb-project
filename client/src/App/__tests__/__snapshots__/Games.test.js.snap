// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the gameList item 1`] = `
<div
<<<<<<< HEAD
  className="game-link"
>
  <a
    href="/game/79335/18"
    onClick={[Function]}
  >
    <p
      className="nes-text is-primary"
    >
      Double Dragon III: The Sacred Stones
    </p>
  </a>
</div>
`;

exports[`renders the saved games component 1`] = `
<div
  className="sc-gzVnrw jnuoKt"
>
  <h1>
    fake@fake.com
=======
  className="sc-EHOje kaDGtO"
>
  <h1>
>>>>>>> d6254eeabc601f61666a349ccec4f094319235ca
    's List of Games
  </h1>
  <div
    className="games-list"
  >
    <div
      className="game-list nes-container is-rounded"
    >
      <h2>
        NES Games
      </h2>
      <div
        className="game-link"
      >
        <a
          href="/game/79335/18"
          onClick={[Function]}
        >
          <p
            className="nes-text is-primary"
          >
            Double Dragon III: The Sacred Stones
          </p>
        </a>
      </div>
    </div>
    <div
      className="game-list nes-container is-rounded"
    >
      <h2>
        SNES Games
      </h2>
    </div>
    <div
      className="game-list nes-container is-rounded"
    >
      <h2>
        64 Games
      </h2>
    </div>
    <div
      className="game-list nes-container is-rounded"
    >
      <h2>
        Game Cube Games
      </h2>
    </div>
  </div>
</div>
`;
